modules/opensearch/includes/opensearch.search.inc

```diff
@@ -147,7 +183,24 @@ function opensearch_search_search(TingSearchRequest $ting_query) {
   // Whether to return all materials from matched collections.
   $options['allObjects'] = $ting_query->getPopulateCollections();
 
-  return new OpenSearchTingSearchResult(ting_do_search($query, $ting_query->getPage(), $ting_query->getCount(), $options), $ting_query);
+  $result = new OpenSearchTingSearchResult(ting_do_search($query, $ting_query->getPage(), $ting_query->getCount(), $options), $ting_query);
+
+  // Handle the situation where opensearch sets a default sort.
+  if (empty($result->getSearchRequest()->getSorts()) && !empty($result->openSearchResult->sortUsed)) {
+    $sort_used = $result->openSearchResult->sortUsed;
+    if ($sort_used === 'rank_main_title') {
+      $sort_used = 'rank_title';
+    }
```

Comment:
The handling of `rank_main_title` seems like a workaround. Perhaps adding a comment would be appropriate?